lib/puppet/provider/file_line/ruby.rb

```diff
@@ -2,6 +2,8 @@
   def exists?
     if resource[:replace].to_s != 'true' and count_matches(match_regex) > 0
       true
+    elsif resource[:replace].to_s != 'false' and count_matches(match_regex) > 1 and resource[:multiple].to_s != 'false'
```

Comment:
one problem i see with this is that :replace defaults to true so any time its not explicitly set, this statement catches it. then, if match is not also set, it fails because `match_regex` would be nil.